src/MainWindow.vala

```diff
@@ -200,15 +200,30 @@ public class MainWindow : Gtk.Window {
                 url_entry
             );
 
-            if (type == WebKit.PolicyDecisionType.NEW_WINDOW_ACTION) {
-                debug ("New window");
-
-                var nav_decision = (WebKit.NavigationPolicyDecision) decision;
-                var uri = nav_decision.navigation_action.get_request ().uri;
-                web_view.load_uri (uri);
+            var nav_decision = (WebKit.NavigationPolicyDecision) decision;
+            string uri = nav_decision.navigation_action.get_request ().uri;
+            switch (type) {
+                case WebKit.PolicyDecisionType.NAVIGATION_ACTION:
+                    if (is_location (uri) && !is_dangerous (uri)) {
+                        decision.use ();
+                    } else if (!is_dangerous (uri)) {
+                        open_externally (uri);
+                    }
+                    decision.ignore ();
+                    return true;
+                case WebKit.PolicyDecisionType.NEW_WINDOW_ACTION:
+                    debug ("New window");
+
+                    if (is_location (uri) && !is_dangerous (uri)) {
```

Comment:
Maybe you just want to remove the `html` mimetypes from the desktop file so you never get a `file://`? Also it shouldn't be "dangerous" really.

Reply:
@TingPing that wouldn't prevent a website from attempting to open a file://, though, which was the actual intent here. There's probably a better way to do that, though.